src/dockerflow/sanic/app.py

```diff
@@ -0,0 +1,306 @@
+        # without pre-configuration. See docs for how to set it up.
+        self.summary_logger = logging.getLogger('request.summary')
+
+        # An ordered dictionary for storing custom Dockerflow checks in.
+        self.checks = OrderedDict()
+
+        # A list of IDs of custom Dockerflow checks to ignore in case they
+        # show up.
+        self.silenced_checks = silenced_checks or []
+
+        # The path where to find the version JSON file. Defaults to the
+        # parent directory of the app root path.
+        self.version_path = version_path
+        self._version_callback = version.get_version
+
+        # Initialize the app if given.
+        if app:
+            self.init_app(app)
+        # Initialize the built-in checks.
+        if redis:
+            self.init_check(checks.check_redis_connected, redis)
+
+    def init_check(self, check, obj):
+        """
+        Adds a given check callback with the provided object to the list
+        of checks. Useful for built-ins but also advanced custom checks.
+        """
+        self.logger.info('Adding extension check %s' % check.__name__)
+        check = functools.wraps(check)(functools.partial(check, obj))
+        self.check(func=check)
+
+    def init_app(self, app):
+        """
+        Add the Dockerflow views and middleware to app.
+        """
+        for uri, name, handler in (
+            ('/__version__', 'version', self._version_view),
+            ('/__heartbeat__', 'heartbeat', self._heartbeat_view),
+            ('/__lbheartbeat__', 'lbheartbeat', self._lbheartbeat_view),
+        ):
+            app.add_route(handler, uri, name=name)
+        app.middleware('request')(self._request_middleware)
+        app.middleware('response')(self._response_middleware)
+        app.exception(Exception)(self._exception_handler)
+
+    def _request_middleware(self, request):
+        """
+        The request middleware.
+        """
+        request['_id'] = str(uuid.uuid4())
+        request['_start_timestamp'] = time.time()
+
+    def _response_middleware(self, request, response):
+        """
+        The response middleware.
+        """
+        if not request.get('_logged'):
+            extra = self.summary_extra(request)
+            self.summary_logger.info('', extra=extra)
+
+    def _exception_handler(self, request, exception):
+        """
+        The exception handler.
+        """
+        extra = self.summary_extra(request)
+        extra['errno'] = 500
+        self.summary_logger.error(str(exception), extra=extra)
+        request['_logged'] = True
+
+    def summary_extra(self, request):
+        """
+        Build the extra data for the summary logger.
+        """
+        out = {
+            'errno': 0,
+            'agent': request.headers.get('User-Agent', ''),
+            'lang': request.headers.get('Accept-Language', ''),
+            'method': request.method,
+            'path': request.path,
+            'uid': '',
```

Comment:
`''` was the default if there was no user logged in via `flask_login`. there's `sanic_auth` but it was taking up too much time to figure out a reasonable way to detect if it was active on the app so i removed support for now.

Reply:
Gotcha, we can pick this up in case auth is needed in a Sanic project then. Would you mind creating a ticket?

Reply:
added support for pulling `uid` using `sanic_auth`

Reply:
er, actually i removed pulling `uid` from `sanic_auth` because it doesn't actually provide auth, it just provides a framework for adding your own auth, and that made it non-trivial to test. I'll open a ticket.